src/main/java/uk/gov/digital/ho/hocs/audit/export/adapter/DateAdapter.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.digital.ho.hocs.audit.export.adapter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static uk.gov.digital.ho.hocs.audit.export.infoclient.ExportViewConstants.*;
+
+
+@Slf4j
+public class DateAdapter implements ExportViewFieldAdapter {
+
+    @Override
+    public String getAdapterType() {
+        return FIELD_ADAPTER_DATE;
+    }
+
+    /**
+     * The aim of this function is to identify dates that are malformed by having a zero in front of the
+     * year, month or day (0yyyy-0mm-0dd). The REGEX used will identify any combination of the zero leading digits.
+     * Once identified, this function converts the date to a format without the leading zero (yyyy-mm-dd).
+     *
+     * @param input The ExportViewFieldAdapter interface supports any kind of object but the calling code only ever passes a string.
+     * @return A string value of the converted (corrected) date or the original value if it is not a date.
+     */
+    @Override
+    public String convert(Object input) {
+        if (input instanceof String) {
+            String date = (String)input;
+            if (date.matches(MALFORMED_DATE_REGEX)) {
```

Comment:
No need for 2 ifs..

This statement should used use the GROUPED_DATE_REGEX (as a compiled pattern) and use find() on that.

Reply:
Implemented static compiled pattern.